### ISSUE TITLE
Create QA test plan for Services autosuggest

### ISSUE THREAD

**Issue by jilladams** [department-of-veterans-affairs/va.gov-cms#20285]
## Description
Create a test plan for verifying Services autosuggest product changes in the Facility Locator.

## Acceptance criteria

- [x] [Facility Locator regression test plan](https://docs.google.com/spreadsheets/d/1wgIM4oIwAQk-E151z07P3hhZuQJ6mhepv7U5tZjdpEI/edit?gid=0#gid=0) is updated with a new tab for Services Autosuggest
    * Added tab: [#15541 Services autosuggest](https://docs.google.com/spreadsheets/d/1wgIM4oIwAQk-E151z07P3hhZuQJ6mhepv7U5tZjdpEI/edit?gid=239023473#gid=239023473) 
- [x] Test cases are added to reflect product changes for Services Autosuggest (@Agile6MSkinner is the SME for this)
- [x] Review it with Eli / Randi to make sure we've caught all behaviors correctly

**Comment by jilladams** [comment]
@Agile6MSkinner suggested a short QA call to work through the basics here. Attendees should be: Michael, Jill, Eli, Randi, Shamia, Jordan, Michelle.

**Comment by randimays** [comment]
For anyone who needs it: https://www.va.gov/data/cms/va-healthcare-services.json Here is the VAMC healthcare services JSON file in production, and a little cheat sheet for how to find the relevant chunks of data that are used in the matching algorithm.

<img width="795" alt="Image" src="https://github.com/user-attachments/assets/500806a0-fef6-444d-aeec-0c147869f704" />

**Comment by jilladams** [comment]
From qa planning meeting: 
1. Good examples of data test cases are here: https://github.com/department-of-veterans-affairs/va.gov-cms/issues/18911#issuecomment-2605309919

2. https://www.va.gov/data/cms/va-healthcare-services.json is the full data set. We can choose a term test case for each of the 6 fields. (Eli noted: this is handled in a unit test. I'm gonna write it out and include it in the
    * Name
    * AKA
    * common conditions
    * API ID (maybe removed from matches, we think)
    * description
    * TRICARE description

3. Handle weighting. Need a test case that reflects a term that should show higher than another match, based on the weighting.

4. Term that should not return a result, for NULL case

5. Add: data test case for a term, location, facility type that we know should return specific facilities.  (MAKE SURE: choose a term that doesn't include ATC data.) Michelle noted: we have a list from Adam of terms that are not in the ATC list.

6. What happens when users type a non-service thing in the services field. Show "No results found" treatment here https://www.figma.com/design/GvNVeG9GAXZaBpny7vNxZP/Facility-Locator?node-id=178-32864&t=sPrphe5AnHcD4fji-0

**Comment by randimays** [comment]
@jilladams For #2: I will be removing the API ID from matches in my working branch when it is unblocked. I also wrote unit tests for the filtering/matching logic that will test all of the code paths. I also plan to add E2E tests to cover the full experience.

**Comment by jilladams** [comment]
Ok! Test plan is updated: https://docs.google.com/spreadsheets/d/1wgIM4oIwAQk-E151z07P3hhZuQJ6mhepv7U5tZjdpEI/edit?gid=239023473#gid=239023473.  

**Service filtering:**
Have added the following test cases (parentheses = the search term to use for testing):
* Name only match (operation)
* AKA only match (corrective)
* Common conditions only match (prostate)
* Description only match (marrow)
* TRICARE description only match (TRICARE)
* CMS only term / not in ATC (mammogram)
* Multi-field/term match (heart & cancer)
* Null match / no results (Invalid)

**UX basics:** 
* Line wrapping when the term name is longer than the dropdown width
* Basic behavior when selecting a term from the list via click or tab: click populates the Services field with the service name and X to close appears

**Search results:**
Added 3 basic test cases using Prod service filters today that we can use to confirm old search / service filter & new search service filter should return the same # of facilities for the same search terms in the same zip code (all using 97206): 
* Primary care
* Mental health
* COVID-19 vaccines

**Comment by jilladams** [comment]
@Shamiaalam for discussion today: For accessibility: our implementation needs to meet the bar set by the 526EZ team in their separate experimental design 

To access that build: 
1. https://staging.va.gov/user-testing/new-conditions/
2. Click "Start your application without signing in"
3. Select an option on 1st screen, Select Continue on 2nd screen, and then you'll see the component on the 3rd screen: "Add new condition"


a11y notes from discussion around 526EZ combo box: https://dsva.slack.com/archives/C8E985R32/p1728398775821319

What that component is already doing in terms of aria labels, voice over behavior etc, our build also needs to do, and we need to include in this test plan.

**Comment by jilladams** [comment]
@Shamiaalam  still working on entering the a11y test cases into the sheet, then will need a quick review.

**Comment by jilladams** [comment]
@Shamiaalam nice! I took a look, and updated some formatting to align with other sections of the test plan. Couple questions / notes: 

1. Row 14: `Announcements: Verify that screen readers announce the input field as a ComboBox and provide updates on its expanded or collapsed state.` -- this one we'll need to figure out with @eselkin since we are not officially using the ComboBox component anymore, but are calling ours autosuggest. I'm not certain what the screen reader will / should say this field is.

2. Row 14: `Live Region: As you type, aria-live region should announce the number of available suggestions dynamically as you type. Number of available suggestions (aria-live="polite"). or (aria-live="assertive ")` --- Can you clarify which aria-live parameter this should use, `polite` or `assertive`?

3. Row 16: `Descriptive Errors: If applicable, using aria-describedby to associate error messages with the input field, also ensuring they are conveyed to assistive technologies. So any error message occurred, screen reader will announce error field.` This is a valid thing to define for the build. We need to figure out 2 things: 
    1. Is this applicable for the build? If so: 
    2. How would a user trigger the condition to get an error, to force the asssistive tech to announce the error? 

I know about a No results found condition (and I've added that to line 16), but I'm not sure there are any other error messages that will apply for this field. This is probably also a question for @eselkin

I've queued this for scrum tomorrow to talk through together, in case it's hard to parse async.

**Comment by jilladams** [comment]
Nice! I updated these things in the [spreadsheet](https://docs.google.com/spreadsheets/d/1wgIM4oIwAQk-E151z07P3hhZuQJ6mhepv7U5tZjdpEI/edit?gid=239023473#gid=239023473) - @Shamiaalam & @eselkin if you could review and just make sure I didn't frame it in some weird way as a test case, and that it makes sense in what we'll be able to actually build.

**Comment by Shamiaalam** [comment]
@jilladams, I will review it after Scrum meeting. Thanks 😊
